cli/tools/registry/mod.rs

```diff
@@ -956,6 +956,23 @@ pub async fn publish(
     bail!("No packages to publish");
   }
 
+  if std::env::var("DENO_TESTING_DISABLE_GIT_CHECK")
+    .ok()
+    .is_none()
+    && !publish_flags.allow_dirty
+    && check_if_git_repo_dirty(cli_options.initial_cwd()).await
+  {
+    if publish_flags.dry_run {
+      log::warn!(
+        "{} Aborting due to --dry-run and uncommitted changes. Check in source code or run with --allow-dirty",
+        colors::yellow("Warning")
+      );
+      return Ok(());
+    } else {
+      bail!("Aborting due to uncommitted changes. Check in source code or run with --allow-dirty");
+    }
```

Comment:
I think this should error only (so moving up the code block from below as-is)? Otherwise someone won't find out about it until going to publish.

Reply:
Done